src/fetch/transfer_file.py

```diff
@@ -362,6 +363,46 @@ def __str__(self) -> str:
         )
 
 
+def map_reward(amount: float, risk_free: bool) -> float:
+    """
+    Converts orderbook rewards based on additional knowledge of "risk_free" transactions
+    """
+    if amount > 0 and risk_free:
+        # Risk Free Orders that are not liquidity orders get 37 COW tokens.
+        return 37.0
+    return amount
+
+
+def aggregate_orderbook_rewards(
+    per_order_df: DataFrame, risk_free_transactions: set[str]
+) -> DataFrame:
+    """
+    Takes rewards per order and adjusts them based on whether they were part of
+    a risk-free settlement or not. After the new amount mapping is complete,
+    the results are aggregated by solver as a sum of amounts and additional
+    "transfer" related metadata is appended. The aggregated dataframe is returned.
+    """
+    per_order_df["amount"] = per_order_df[["amount", "tx_hash"]].apply(
+        lambda x: map_reward(x.amount, risk_free=x.tx_hash in risk_free_transactions),
+        axis=1,
+    )
+    result_agg = (
+        per_order_df.groupby("solver")["amount"].agg(["count", "sum"]).reset_index()
+    )
+    del per_order_df  # We don't need this anymore!
+    # Add token address to each column
+    result_agg["token_address"] = "0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB"
+    # Rename columns to align with "Transfer" Object
+    result_agg = result_agg.rename(
+        columns={"sum": "amount", "count": "num_trades", "solver": "receiver"}
+    )
+    # Convert float amounts to WEI
+    result_agg["amount"] = result_agg["amount"].apply(
```

Comment:
Just clarifying this. Are you converting the COW rewards into ETH as well?

Reply:
No just to wei. As in 18 decimals

Reply:
The ether string is just the keyword for 18 decimals.

Reply:
Yes, good. A bit confused since these amounts are in COW, and it's not clear to me where you pass this information to this `toWei()` function. Probably I am missing something here since I don't fully get the code here.

Reply:
We don't pass any token information here because COW token has 18 decimals (which is analogous to "ether") the key word used in `web3.toWEI`. This method only turns the COW reward amounts into WEI.

If you are referring to the "token_address" column here, that is because the results returns here are "transfer objects"

They require `receiver, token_address, amount (in WEI)`